src/components/PeopleFilters.tsx

```diff
@@ -1,18 +1,133 @@
+  setSearchWith,
+  currentSearchParams,
+  setSearchParams,
+}: {
+  setSearchWith: Search;
+  currentSearchParams: URLSearchParams;
+  setSearchParams: SetURLSearchParams;
+}) => {
+  const [searchParamSecondAttach] = useSearchParams();
+  const centuriesAll = currentSearchParams.getAll('centuries') || [];
+  const sexAll = currentSearchParams.get('sex') || '';
+  const searchAll = currentSearchParams.get('search');
+  const centuriesArray = ['16', '17', '18', '19', '20'];
+
+  const onInput = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const sendForm: {
+      search: string,
+      centuries: string[],
+      sex?: string,
+    } = {
+      search: event.target.value,
+      centuries: [...centuriesAll],
+    };
+
+    if (sexAll !== '') {
+      sendForm.sex = sexAll;
+    }
+
+    setSearchParams(sendForm);
+
+    if (event.target.value === '') {
+      setSearchParams('');
+    }
+  };
+
+  const handleClick = (value?: {
+    sex?: string | null;
+    centuries?: string | null;
+  }) => {
+    if (value?.centuries) {
+      const decision = {
+        search: setSearchWith(searchParamSecondAttach, {
+          search: searchAll,
+          centuries: centuriesAll.includes(value.centuries)
+            ? centuriesAll.filter((l) => l !== value.centuries)
+            : [...centuriesAll, value.centuries],
+          sex: sexAll === '' ? null : sexAll,
+        }),
+      };
+
+      return decision;
+    }
+
+    if (value?.centuries === null) {
+      const decision = {
+        search: setSearchWith(searchParamSecondAttach, {
+          search: searchAll === '' ? null : searchAll,
+          centuries: null,
+          sex: sexAll === '' ? null : sexAll,
+        }),
+      };
+
+      return decision;
+    }
+
+    if (value?.sex || value?.sex === null) {
+      const decision = {
+        search: setSearchWith(searchParamSecondAttach, {
+          search: searchAll === '' ? null : searchAll,
+          centuries: [...centuriesAll],
+          sex: value.sex,
+        }),
+      };
+
+      return decision;
+    }
+
+    return '/people';
+  };
+
   return (
     <nav className="panel">
       <p className="panel-heading">Filters</p>
 
       <p className="panel-tabs" data-cy="SexFilter">
-        <a className="is-active" href="#/people">All</a>
-        <a className="" href="#/people?sex=m">Male</a>
-        <a className="" href="#/people?sex=f">Female</a>
+        <Link
+          className={classNames({
+            'is-active': sexAll?.length === 0,
+          })}
+          to={handleClick({ sex: null })}
+        >
+          All
+        </Link>
+        <Link
+          className={classNames({
+            'is-active': sexAll?.includes('m'),
+          })}
+          to={handleClick({ sex: 'm' })}
+        >
+          Male
+        </Link>
+        <Link
+          className={classNames({
+            'is-active': sexAll?.includes('f'),
+          })}
+          to={handleClick({ sex: 'f' })}
```

Comment:
Consider will be better if you create a variable for all magic wods and use them everywhere

Reply:
Programm have loader.

Reply:
https://github.com/mate-academy/react_people-table-advanced/assets/101456317/c09f9a01-b28d-4a90-908d-e68855740314

Reply:
add frame, like in [working example](https://mate-academy.github.io/react_people-table-advanced/)